common/core/desktop/src/kmx/kmx_processor.cpp

```diff
@@ -240,3 +329,45 @@ km_kbp_keyboard_key * kmx_processor::get_key_list() const  {
   rules[n] =  KM_KBP_KEYBOARD_KEY_LIST_END;
   return rules;
 }
+
+km_kbp_keyboard_imx * kmx_processor::get_imx_list() const  {
+
+  const uint32_t store_cnt = _kmx.GetKeyboard()->Keyboard->cxStoreArray;
+  const LPSTORE store_array = _kmx.GetKeyboard()->Keyboard->dpStoreArray;
+  uint16_t fn_count = 0;
+  uint16_t fn_idx = 0;
+
+  for(uint32_t i = 0; i < store_cnt; i++) {
+    LPSTORE p_store = &store_array[i];
+    if(p_store->dwSystemID == TSS_CALLDEFINITION) {
+      fn_count++;
+    }
+  }
+
+  km_kbp_keyboard_imx *imx_list = new km_kbp_keyboard_imx[fn_count + 1];
+
+  for(uint32_t i = 0; i < store_cnt; i++) {
+    LPSTORE p_store = &store_array[i];
+    if(p_store->dwSystemID == TSS_CALLDEFINITION) {
+			/* Break the store string into components */
+
+			PKMX_WCHAR full_fn_name = p_store->dpString;
+      PKMX_WCHAR pt = NULL;
+      PKMX_WCHAR lib_name = u16tok(full_fn_name, u':', &pt);
+			PKMX_WCHAR fn_name = u16tok(NULL, u':', &pt);
+
+			if(!lib_name || !fn_name){
+				continue;
+			}
+
+      imx_list[fn_idx].library_name = lib_name;
+      imx_list[fn_idx].function_name = fn_name;
+      imx_list[fn_idx].imx_id = i;
+      fn_idx++;
+		}
+  }
```

Comment:
`u16tok` is good. Eliminates a memory leak too, but introduces a little problem. This is going to fail if we call `get_imx_list` more than once. In the original Windows core, the underlying store `p_store->dpString` is not mutated. Now, as the string is copied, that doesn't hold here.

(The mutation to the structure that does occur in the old Windows core is to mark call definitions that are invalid or where the load fails, as `TSS_CALLDEFINITION_LOADFAILED`, which avoids trying to load them again.)

We probably need to copy the string, and then free it for each entry, in `km_kbp_keyboard_imx_list_dispose` (which means more internal plumbing and perhaps a bit of internal knowledge crossing the core/kmx keyboardprocessor boundary, ugh). Or something like that?